### PR TITLE
feature: generate gensort data in 1GB partitions

### PR DESCRIPTION
**Pull request by rahij**

### CODE REVIEW

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java [hunk 1]
```diff
@@ -106,47 +141,49 @@ public void generate() {
...
+                    paths.csvDir(scale),
+                    dataDir.toFile(),
+                    MoreExecutors.newDirectExecutorService());
+        });
     }
 
-    private int estimateNumRecords(int scale) {
-        return (scale * 1024 * 1024) / BYTES_PER_RECORD;
+    private long estimateNumRecords(int scale) {
+        return (scale * 1024 * 1024 * 1024) / BYTES_PER_RECORD;
```
Comment (rahij) [inline spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java]: bug :(

Comment (lwwmanning) [reply spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java]: but you found it :)

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/DefaultParquetTransformer.java [hunk 2]
```diff
@@ -24,13 +26,22 @@
...
+            Set<String> sourcePaths,
+            String destinationPath,
+            String delimiter) {
+        Optional<Dataset<Row>> unioned = sourcePaths.stream()
+                .map(path -> sparkSession
+                        .read()
+                        .format("csv")
+                        .option("delimiter", delimiter)
+                        .schema(schema)
+                        .load(path))
```
Comment (lwwmanning) [inline spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/DefaultParquetTransformer.java:40]: IIRC, doesn't `load` have a version that takes multiple files...?

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java [hunk 3]
```diff
@@ -41,6 +48,9 @@
     private static final Logger log = LoggerFactory.getLogger(GenSortDataGenerator.class);
 
     private static final int BYTES_PER_RECORD = 100;
+    private static final long BYTES_PER_GB = 1024 * 1024 * 1024;
```
Comment (lwwmanning) [inline spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java:53]: fine to do like this, but a handy thing to be aware of for this sort of thing: https://github.com/palantir/human-readable-types/blob/develop/human-readable-types/src/main/java/com/palantir/humanreadabletypes/HumanReadableByteCount.java

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java [hunk 4]
```diff
@@ -41,6 +48,9 @@
...
+    private static final long PARTITION_SIZE = BYTES_PER_GB; // 1GB
+    private static final long RECORDS_PER_CHUNK = PARTITION_SIZE / BYTES_PER_RECORD;
```
Comment (lwwmanning) [inline spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java]: `RECORDS_PER_PARTITION`

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java [hunk 5]
```diff
@@ -106,47 +142,49 @@ public void generate() {
...
+            int returnCode;
+            try {
+                returnCode = gensortProcess.waitFor();
+            } catch (InterruptedException e) {
+                throw new SafeRuntimeException("Thread was interrupted while waiting for gensort process", e);
+            }
+            if (returnCode != 0) {
+                throw new IllegalStateException(String.format("genSort failed with return code %d", returnCode));
+            }
+            log.info("Finished running gensort");
```
Comment (lwwmanning) [inline spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java]: would be good to include info about scale/partition/etc

**Review by lwwmanning — APPROVED** [review]
left a couple comments, can be done in a FLUP

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/DefaultParquetTransformer.java (hunk 2 above)
Comment (rahij) [reply spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/DefaultParquetTransformer.java:40]: Yeah but we need to convert to a scala seq - not sure we want to add a scala dep for that?

**Pull request merged**